tests/ui/self_named_constructor.rs

```diff
@@ -0,0 +1,50 @@
+#![warn(clippy::self_named_constructor)]
+
+struct Q;
```

Comment:
Nitpicking, but still: it's hard to read tests with one-letter identifiers. Using something like `ShouldSpawn` /  `ShouldNotSpawn` will be much more helpful.

UI tests are not supposed to be read very often, but usually, if you have to read them, you've got some trouble to deal with. And for those cases, it's better to keep tests readable.

Reply:
Good point, I will make the tests easier to understand